server/src/main/java/org/elasticsearch/index/mapper/AbstractGeometryFieldMapper.java

```diff
@@ -85,6 +85,11 @@ protected AbstractGeometryFieldType(
             this.geometryParser = geometryParser;
         }
 
+        @Override
+        public final boolean mayExistInIndex(SearchExecutionContext context) {
```

Comment:
I was thinking we could just remove these overrides for all the non-text fields like this? They already return `TextSearchInfo.NONE` so they'll be skipped. That way the `mayExistInIndex` method checks exactly what it claims, whether the field exists at all.